packages/secretlint/test/create-secretlintrc.test.ts

```diff
@@ -0,0 +1,67 @@
+// LICENSE : MIT
+"use strict";
+import assert from "assert";
+import path from "path";
+import fs from "fs";
+import os from "os";
+import { runConfigCreator } from "../src/create-secretlintrc";
+
+/**
+ *
+ * Create .secretlintrc.json
+ * - yarn test -g "create-secretlintrc-test"
+ *
+ * File Overview
+ *
+ * - tmpConfigDir
+ *   - A temporal directory with "secretlint-config"
+ *   - Create .secretlintrc.json on the directory
+ * - packageFilePath
+ *   - A path of package.json to use of this test
+ *   - When using this package.json, copy it to tmpConfigDir
+ * - expectedConfigFilePath
+ *   - A path of .secretlintrc.json to create by this test
+ */
+describe("create-secretlintrc-test", function () {
+    let tmpConfigDir: string;
+
+    beforeEach(function () {
+        tmpConfigDir = fs.mkdtempSync(path.join(os.tmpdir(), "secretlint-config"));
+        const packageFilePath = path.join(__dirname, "fixtures", "package.json");
+        fs.promises.copyFile(packageFilePath, tmpConfigDir + "/package.json");
+    });
+
+    afterEach(function () {
+        fs.rmSync(tmpConfigDir, { recursive: true });
+    });
+
+    context("when pacakge.json has @secretlint/* packages", function () {
+        it("Run secretlint --init", async function () {
+            const expectedConfigFilePath = `${tmpConfigDir}/.secretlintrc.json`;
+            return runConfigCreator({ cwd: tmpConfigDir }).then(function (actual) {
+                assert.equal(actual.exitStatus, 0);
+                assert.equal(actual.stdout, `Create ${expectedConfigFilePath}`);
+                assert.equal(actual.stderr, null);
+            });
+        });
+    });
+
+    context("when .secretlintrc.json is already existed", function () {
+        it("should be an error", async function () {
+            const expectedConfigFilePath = `${tmpConfigDir}/.secretlintrc.json`;
+            return runConfigCreator({ cwd: tmpConfigDir })
+                .then((actual) => {
+                    assert.equal(actual.exitStatus, 0);
+                    assert.equal(actual.stdout, `Create ${expectedConfigFilePath}`);
```

Comment:
Note:
https://nodejs.org/api/assert.html#assertmatchstring-regexp-message
assert.match may be useful for matching text.

Reply:
@azu 
I have a question.
Which is better assert.strictEqual() or assert.match()? 
Could you share with me your idea to choose two assert methods?

In the following case, we choose assert.match().
https://github.com/secretlint/secretlint/pull/306/commits/72f1a651989a95044d63633daa89a4fa07393e67

However, in some other cases, we choose assert.strictEqual().

Reply:
`assert.strictEqual()` is string matching.
`assert.match()` is regexp matching.

Which is appropriate is a case-by-case basis.

For example, if you **only** want to test if the result is an error, `assert.match()` is useful.

```js
const validatePrimitiveValue = (o) => {
   if(typeof o === "object" && o !== null) {
     return { ok: false, message: "Error: this is object" }
   }
   if(typeof o === "function") {
     return { ok: false, message: "Error: this is function" }
   }
   return { ok: true }
}
const result = validateSome({ /* somthing */ });
assert.match(result.message, /Error:.*/);
```

If you write this in `assert.strictEqual`, it becomes a little confusing and less assertion error details.

```js
const validatePrimitiveValue = (o) => {
   if(typeof o === "object" && o !== null) {
     return { ok: false, message: "Error: this is object" }
   }
   if(typeof o === "function") {
     return { ok: false, message: "Error: this is function" }
   }
   return { ok: true }
}
const result = validateSome({ /* somthing */ });
const isErrorMessage = /Error:.*/.test(result.message);
assert.strictEqual(isErrorMessage, true);
```

In most case, `assert.strictEqual` is best and simple as visual.
However, It has the disadvantage of being inflexible.